packages/flutter/test/material/snack_bar_test.dart

```diff
@@ -1146,4 +1053,283 @@ void main() {
+
+          expect(initialBottomLeft, finalBottomLeft);
+          expect(initialBottomRight, finalBottomRight);
+        },
+      );
+    }
+
+    testWidgets(
+      '${SnackBarBehavior.fixed} should align SnackBar with the bottom of Scaffold '
+      'when Scaffold has a FloatingActionButton',
+      (WidgetTester tester) async {
+        await tester.pumpWidget(
+          MaterialApp(
+            home: Scaffold(
+              body: Container(),
+              floatingActionButton: FloatingActionButton(onPressed: () {}),
+            ),
+          ),
+        );
+
+        final ScaffoldState scaffoldState = tester.state(find.byType(Scaffold));
+        scaffoldState.showSnackBar(
+          const SnackBar(
+            content: Text('Snackbar text'),
+            behavior: SnackBarBehavior.fixed,
+          ),
+        );
+
+        await tester.pump();
+
+        final Offset snackBarBottomRight = tester.getBottomRight(find.byType(SnackBar));
+        final Offset scaffoldBottomRight = tester.getBottomRight(find.byType(Scaffold));
+
+        expect(snackBarBottomRight, equals(scaffoldBottomRight));
+
+        final Offset snackBarBottomLeft = tester.getBottomLeft(find.byType(SnackBar));
+        final Offset scaffoldBottomLeft = tester.getBottomLeft(find.byType(Scaffold));
+
+        expect(snackBarBottomLeft, equals(scaffoldBottomLeft));
+      },
+    );
+
+    testWidgets(
+      '${SnackBarBehavior.floating} should align SnackBar with the top of FloatingActionButton'
+      'when Scaffold has a FloatingActionButton',
+      (WidgetTester tester) async {
+        await tester.pumpWidget(MaterialApp(
+          home: Scaffold(
+            floatingActionButton: FloatingActionButton(
+              child: const Icon(Icons.send),
+              onPressed: () {},
+            ),
+            body: Builder(
+              builder: (BuildContext context) {
+                return GestureDetector(
+                  onTap: () {
+                    Scaffold.of(context).showSnackBar(SnackBar(
+                      content: const Text('I am a snack bar.'),
+                      duration: const Duration(seconds: 2),
+                      action: SnackBarAction(label: 'ACTION', onPressed: () {}),
+                      behavior: SnackBarBehavior.floating,
+                    ));
+                  },
+                  child: const Text('X'),
+                );
+              }
+            ),
+          ),
+        ));
+        await tester.tap(find.text('X'));
+        await tester.pump(); // start animation
+        await tester.pump(const Duration(milliseconds: 750)); // Animation last frame.
+
+        final Offset snackBarBottomCenter = tester.getBottomLeft(find.byType(SnackBar));
+        final Offset floatingActionButtonTopCenter = tester.getTopLeft(
+          find.byType(FloatingActionButton));
+
+        // Since padding and margin is handled inside snackBarBox,
```

Comment:
What is `snackBarBox`? If it's referring to the `RenderBox` of the SnackBar, maybe generalizing the term might be better:
```suggestion
        // Since padding between the SnackBar and the FAB is created by the SnackBar,
        // the bottom offset of the SnackBar should be equal to the top offset of the FAB.
```

Edit: I realized later that this piece was copy-pasted, but this is a good opportunity to clean this up :)

Reply:
Fixed.

Reply:
Fixed.